### ISSUE TITLE
Can't change language in app

### ISSUE THREAD

**Issue by SnIPeRSnIPeR** [nicotine-plus/nicotine-plus#1796]
Hello! Can you help me please? I can't find the option to change language from English to another in app... How to change it right way? Thank you!

**Comment by slook** [comment]
Nicotine+ will try to autodetect your language based on what locale your system is using. Nicotine+ can be forced to use another language, you can do this by specifying a locale before starting Nicotine+ from a command line, e.g.:

- English: `LANGUAGE=en_US.UTF-8 python3 nicotine`
- French: `LANGUAGE=fr_FR.UTF-8 python3 nicotine`

What is your OS and installation method?

**Comment by SnIPeRSnIPeR** [comment]
I am using Windows 10 and "Official Build: 64-bit Installer [SHA256]" from https://nicotine-plus.org/doc/DOWNLOADS.html#windows

How can I change language from EN to another for example?

**Comment by mathiascode** [comment]
It's possible that translations are currently broken in the Windows version of Nicotine+. What's your system language? (we don't have a GUI option for selecting the language at the moment)

**Comment by SnIPeRSnIPeR** [comment]
Hmm, very bad... I need Russian language.

**Comment by slook** [comment]
The cause of the problem is not known as yet, since Windows testing is limited it might take a while to figure out.

In the meantime, you could try installing Git and running Nicotine+ from source, see: https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#git

Then, it might be possible for you to apply the language from the translation source following these instructions: https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TRANSLATIONS.md#testing-updated-translations

This solution is not ideal, further replies will be added to this thread when it is properly fixed. Thank you for reporting the issue.

**Comment by hboetes** [comment]
Just like you would expect, actually:

Open a cmd window, type `set LANGUAGE=ru_RU.UTF-8` and then go into the `Nicotine+` folder and type `"Nicotine+.exe`

There are many guides online on how to permanently set an environment variable in windows, which will make live even easier.

![image](https://user-images.githubusercontent.com/2285225/148107490-59fdd30b-160c-4946-97cc-a41fde54168b.png)

**Comment by hboetes** [comment]
@mathiascode quite a few items in various translations for windows are missing, as you can see from the screenshot, the same items were missing in other languages as well.

**Comment by SnIPeRSnIPeR** [comment]
> @mathiascode quite a few items in various translations for windows are missing, as you can see from the screenshot, the same items were missing in other languages as well.

Thanks! It works, but a lot of thing is still in english. How to manually replace files from https://hosted.weblate.org/projects/nicotine-plus/#languages to change the language in app? Is it possible?

**Comment by hboetes** [comment]
> > @mathiascode quite a few items in various translations for windows are missing, as you can see from the screenshot, the same items were missing in other languages as well.
> 
> Thanks! It works, but a lot of thing is still in english. How to manually replace files from https://hosted.weblate.org/projects/nicotine-plus/#languages to change the language in app? Is it possible?

That won't help, like I said: it's a bug, we're going to look at that and fix it. Please bear with us, and thanks for your understanding.

**Comment by mathiascode** [comment]
What needs to be done:
- Set the LANGUAGE/LANG/LC_ALL variable on startup if Windows or macOS is detected, since it's not done automatically
- Investigate why GtkBuilder (ui) files are not translated on Windows and macOS

**Comment by hboetes** [comment]
Closed in #1799 =)

**Comment by slook** [comment]
@SnIPeRSnIPeR please can you download and install the latest Nicotine+ 3.2.1.dev1 (Testing) from the following link, and verify if the problem is resolved on your system? It should automatically detect language, and the strings for the missing things should be fixed now:

- **https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#windows**

> Closed in #1799 =)

There is still an issue with macOS that the language needs to be established explicitly. Perhaps it is worth keeping this issue open until that is done.

**Comment by hboetes** [comment]
> There is still an issue with macOS that the language needs to be established explicitly. Perhaps it is worth keeping this issue open until that is done.

That's a variation on feature creep; issue creep. :-)

**Comment by SnIPeRSnIPeR** [comment]
> @SnIPeRSnIPeR please can you download and install the latest Nicotine+ 3.2.1.dev1 (Testing) from the following link, and verify if the problem is resolved on your system? It should automatically detect language, and the strings for the missing things should be fixed now:
> 
> * **https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#windows**
> 
> > Closed in #1799 =)
> 
> There is still an issue with macOS that the language needs to be established explicitly. Perhaps it is worth keeping this issue open until that is done.

Wow, it works now on Windows 10 :)

Can anyone change this text in main menu between themselves? I think this translation is made by non-russian person. Right now "Uploads" is "Downloads" and vica versa in russian translation.

![2022-01-06_05-36-30](https://user-images.githubusercontent.com/31162059/148299997-3ff82a04-d499-43aa-8705-3962400ea020.png)

**Comment by hboetes** [comment]
Read this: https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TRANSLATIONS.md
And you're good to go and fix all problems you find over here: https://hosted.weblate.org/projects/nicotine-plus/nicotine-plus/ru/

**Comment by SnIPeRSnIPeR** [comment]
I can't find this one in translation:

![2022-01-06_05-54-28](https://user-images.githubusercontent.com/31162059/148301344-a111076b-00c0-4241-a60f-6e72113c3c8d.png)

**Comment by mathiascode** [comment]
> I can't find this one in translation:

It should be available now.

**Comment by Mehavoid** [comment]
> Wow, it works now on Windows 10 :)
> 
> Can anyone change this text in main menu between themselves or where can I change it by myself? I think this translation is made by non-russian person. Right now "Uploads" is "Downloads" and vica versa in russian translation.
> 
> ![2022-01-06_05-36-30](https://user-images.githubusercontent.com/31162059/148299997-3ff82a04-d499-43aa-8705-3962400ea020.png)

At some point I was confused. That should be fixed now!

**Comment by SnIPeRSnIPeR** [comment]
> At some point I was confused. That should be fixed now!

Unfortunately, it didn't fixed yet in latest developer version.

**Comment by slook** [comment]
It will be after pull request https://github.com/nicotine-plus/nicotine-plus/pull/1800 is merged.

**Comment by SnIPeRSnIPeR** [comment]
> It will be after pull request #1800 is merged.

Thanks!

**Comment by SnIPeRSnIPeR** [comment]
This text in english at the latest dev version...

![2022-01-07_03-43-56](https://user-images.githubusercontent.com/31162059/148449247-f53d7711-decd-4f8c-9113-b5c32a0fbcf4.png)

**Comment by mathiascode** [comment]
Something in https://github.com/nicotine-plus/nicotine-plus/pull/1801 broke the translations for .ui files, but I'm not sure what's wrong yet.

**Comment by mathiascode** [comment]
gettext 0.21 was pushed to MSYS2 earlier today (around the same time that PR was merged), which broke something. Investigating.

**Comment by mathiascode** [comment]
Translations should be working properly on Windows and macOS in the latest unstable build.

**Comment by hboetes** [comment]
I just tested it, looks like it should. Thanks, @mathiascode !

**Comment by slook** [comment]
There are many international users running Stable who may be affected by this issue, perhaps it is worth pinning until the next release @mathiascode ?

**Comment by mathiascode** [comment]
I'm surprised this wasn't spotted earlier, but perhaps this was due to our incomplete translations until recently. Translations haven't worked properly in Windows or macOS for well over a year now.

**Comment by SnIPeRSnIPeR** [comment]
> Translations should be working properly on Windows and macOS in the latest unstable (testing) build.

Working good, but some things is still in English
![2022-01-08_02-02-52](https://user-images.githubusercontent.com/31162059/148593870-5106e54d-856b-429c-a2ba-d0d369301f77.png)

**Comment by mathiascode** [comment]
That string was marked as "Needs editing" until recently, and it takes some time before the Weblate changes are merged into Nicotine+.

**Comment by hboetes** [comment]
@SnIPeRSnIPeR 
Go ahead and fix the missing strings: :-)
 https://hosted.weblate.org/translate/nicotine-plus/nicotine-plus/ru/?q=state:needs-editing

**Comment by slook** [comment]
@SnIPeRSnIPeR Thank you for being so thorough with your contributions, everybody appreciates your brilliant work :)

**Comment by SnIPeRSnIPeR** [comment]
> That string was marked as "Needs editing" until recently, and it takes some time before the Weblate changes are merged into Nicotine+.

Ok. Got it :)

> Go ahead and fix the missing strings: :-)
> https://hosted.weblate.org/translate/nicotine-plus/nicotine-plus/ru/?q=state:needs-editing

Done, but translated version is not perfect. Will try to fix it.

> @SnIPeRSnIPeR Thank you for being so thorough with your contributions, everybody appreciates your brilliant work :)
:)

**Comment by hboetes** [comment]
> Done, but translated version is not perfect. Will try to fix it.

Perfect translations don't exist, perfect wording doesn't exist. All we can do is our best.

Thanks for your effort!

**Comment by hboetes** [comment]
@Mehavoid @SnIPeRSnIPeR 
Oh yes, how would you guys like to be mentioned in the TRANSLATORS.md and the credits list in Help → About?

**Comment by SnIPeRSnIPeR** [comment]
> Oh yes, how would you guys like to be mentioned in the TRANSLATORS.md and the credits list in Help → About?

Yes, will be great! You can add me like "SnIPeRSnIPeR"
